src/Subtitles/index.js

```diff
@@ -0,0 +1,174 @@
+      this._captions &&
+      this._captions.length &&
+      this._currentSubtitle &&
+      this._nextSubtitle &&
+      Number(currentTime.toFixed(0)) < Number(this._nextSubtitle.end.toFixed(0)) &&
+      Number(currentTime.toFixed(0)) >= Number(this._currentSubtitle.start.toFixed(0))
+    ) {
+      if (
+        Number(currentTime.toFixed(0)) >= Number(this._currentSubtitle.start.toFixed(0)) &&
+        Number(currentTime.toFixed(0)) < Number(this._currentSubtitle.end.toFixed(0))
+      ) {
+        return this._currentSubtitle.payload
+      } else if (
+        Number(currentTime.toFixed(0)) >= Number(this._nextSubtitle.start.toFixed(0)) &&
+        Number(currentTime.toFixed(0)) < Number(this._nextSubtitle.end.toFixed(0))
+      ) {
+        return this._nextSubtitle.payload
+      } else {
+        return ''
+      }
+    } else {
+      updateSubtitles()
+    }
+
+    function updateSubtitles() {
+      // updates current and next subtitle text values
+      if (self._captions && self._captions.length) {
+        if (
+          Number(currentTime.toFixed(0)) <=
+          Number(self._captions[self._captions.length - 1].start.toFixed(0))
+        ) {
+          if (Number(currentTime.toFixed(0)) < Number(self._captions[0].end.toFixed(0))) {
+            if (self._captions[1] && self._captions[1].payload) {
+              self._nextSubtitle = self._captions[1]
+            }
+            self._currentSubtitle = self._captions[0]
+          } else {
+            for (let i = 0; i < self._captions.length; i++) {
+              if (Number(self._captions[i].start.toFixed(0)) >= Number(currentTime.toFixed(0))) {
+                self._captions[i + 1] && self._captions[i + 1].payload
+                  ? (self._nextSubtitle = self._captions[i + 1])
+                  : { payload: '' }
+                self._currentSubtitle = self._captions[i]
+                break
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  // parses subtitle file and returns list of time, text objects
+  static parseSubtitles(plainSub) {
+    let linesArray = plainSub
+      .trim()
+      .replace('\r\n', '\n')
+      .split(/[\r\n]/)
+      .map(line => {
+        return line.trim()
+      })
+    let cues = []
+    let start = null
+    let end = null
+    let payload = null
+    let lines = linesArray.filter(item => item !== '' && isNaN(item))
+    linesArray = []
+    for (let i = 0; i < lines.length; i++) {
+      if (lines[i].indexOf('-->') >= 0) {
+        let splitted = lines[i].split(/[ \t]+-->[ \t]+/)
+
+        start = Subtitles.parseTimeStamp(splitted[0])
+        end = Subtitles.parseTimeStamp(splitted[1])
+      } else if (lines[i] !== '') {
+        if (start && end) {
+          if (i + 1 < lines.length && lines[i + 1].indexOf('-->') >= 0) {
+            let cue = new SubtitleAsset({
+              start,
+              end,
+              payload: payload ? payload + ' ' + lines[i] : lines[i],
+            })
```

Comment:
Is there a benefit to having these subtitles as a Class? They could be a generic object like `{ start: 123, end: 456, lines: ['Line 1', 'Line 2'] }`, right?